File: EXAMEN/01.SERVIDOR/WS_JAVA_REST_BanQuito/src/main/java/ec/edu/monster/util/PasswordUtil.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.monster.util;

/**
 *
 * @author danie
 */

import java.nio.charset.StandardCharsets;
import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;

public class PasswordUtil {

    public static String hashPassword(String plain) {
        if (plain == null) return null;
        try {
            MessageDigest md = MessageDigest.getInstance("SHA-256");
            byte[] bytes = md.digest(plain.getBytes(StandardCharsets.UTF_8));
            return bytesToHex(bytes);
        } catch (NoSuchAlgorithmException e) {
            throw new RuntimeException("Error al hashear contraseña", e);
        }
    }

    public static boolean matches(String plain, String hash) {
        if (plain == null || hash == null) return false;
        String calculated = hashPassword(plain);
        return calculated.equals(hash);
    }

    private static String bytesToHex(byte[] bytes) {
        StringBuilder sb = new StringBuilder(bytes.length * 2);
        for (byte b : bytes) {
            sb.append(String.format("%02x", b));
        }
        return sb.toString();
    }
}

File: PROYECTO/01.SERVIDOR/WS_DOTNET_SOAP_Comercializadora/ec.edu.monster.utils/PasswordUtil.cs
using System;
using System.Security.Cryptography;
using System.Text;

namespace WS_DOTNET_SOAP_Comercializadora.ec.edu.monster.utils
{
    public static class PasswordUtil
    {
        public static string HashPassword(string plain)
        {
            if (plain == null) return null;

            using (var sha = SHA256.Create())
            {
                byte[] bytes = Encoding.UTF8.GetBytes(plain);
                byte[] hashBytes = sha.ComputeHash(bytes);

                var sb = new StringBuilder(hashBytes.Length * 2);
                foreach (byte b in hashBytes)
                {
                    sb.AppendFormat("{0:x2}", b); // hex minúsculas
                }
                return sb.ToString();
            }
        }

        public static bool Matches(string plain, string hash)
        {
            if (string.IsNullOrEmpty(plain) || string.IsNullOrEmpty(hash))
                return false;

            string calculated = HashPassword(plain);
            return string.Equals(calculated, hash, StringComparison.OrdinalIgnoreCase);
        }
    }
}
